pkg/build/golang/deps.go

```diff
@@ -0,0 +1,83 @@
+package golang
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/pkg/archive"
+)
+
+func parseDependencies(raw string) map[string]string {
+	rawModules := strings.Split(raw, "\n")
+	modules := map[string]string{}
+
+	for _, module := range rawModules {
+		module = strings.TrimSpace(module)
+		if module == "" {
+			continue
+		}
+
+		parts := strings.Split(module, " ")
+		path := strings.TrimSpace(parts[0])
+		version := ""
+
+		if len(parts) == 2 {
+			version = strings.TrimSpace(parts[1])
+		}
+
+		modules[path] = version
+	}
+
+	return modules
+}
+
+func parseDependenciesFromDocker(cli *client.Client, imageID string) (map[string]string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
```

Comment:
We have a `ctx` one level up - maybe worth including `ctx` as an argument to this function, so that all container actions here get canceled if the ctx in the argument gets canceled.